front/lib/api/assistant/agent.ts

```diff
@@ -409,3 +272,184 @@ export async function* runAgent(
     message: agentMessage,
   };
 }
+
+async function pickAction(
+  auth: Authenticator,
+  configuration: AgentConfigurationType,
+  conversation: ConversationType
+): Promise<AgentActionConfigurationType | null> {
+  // TODO(@fontanierh): This is a placeholder for the action selection logic.
+  // This will be replaced by the multi-actions "pick tool" logic.
+  // This should also include the inputs-generation logic (generating the arguments for the action)
+  void auth;
+  void conversation;
+
+  const action = deprecatedGetFirstActionConfiguration(configuration);
+
+  if (action !== null) {
+    return action;
+  }
+
+  return null;
+}
+
+async function* runAction(
+  auth: Authenticator,
+  configuration: AgentConfigurationType,
+  conversation: ConversationType,
+  userMessage: UserMessageType,
+  agentMessage: AgentMessageType,
+  action: AgentActionConfigurationType
+): AsyncGenerator<
+  AgentErrorEvent | AgentActionEvent | AgentActionSuccessEvent,
+  void
+> {
+  if (isRetrievalConfiguration(action)) {
```

Comment:
Can we create one function per action type?

Reply:
Yes, but I'd argue this should rather be done when we refactor actions into classes (which is planned). We likely want `action.run()`